query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/PinotBasedRequestHandler.java

```diff
@@ -147,6 +185,130 @@ public QueryCost canHandle(
+      }
+
+      // Assert the values now.
+      return isEquals(viewColumnFilter.getValues(), queryFilter.getRhs());
+    }
+  }
+
+  /**
+   * Checks that the values from the given expression are a subset of the given set.
+   */
+  private boolean isSubSet(Set<String> values, Expression expression) {
+    if (!expression.hasLiteral()) {
+      return false;
+    }
+
+    return values.containsAll(getExpressionValues(expression.getLiteral()));
+  }
+
+  /**
+   * Checks that the values from the given expression are a subset of the given set.
+   */
+  private boolean isEquals(Set<String> values, Expression expression) {
+    if (!expression.hasLiteral()) {
+      return false;
+    }
+
+    return values.equals(getExpressionValues(expression.getLiteral()));
+  }
+
+  private Set<String> getExpressionValues(LiteralConstant literalConstant) {
+    Set<String> expressionValues = new HashSet<>();
+    Value value = literalConstant.getValue();
+    switch (value.getValueType()) {
+      case STRING:
+        expressionValues.add(value.getString());
+        break;
+      case STRING_ARRAY:
+        expressionValues.addAll(value.getStringArrayList());
+        break;
+      case INT:
+        expressionValues.add(String.valueOf(value.getInt()));
+        break;
+      case INT_ARRAY:
+        expressionValues.addAll(value.getIntArrayList().stream().map(Object::toString).collect(
+            Collectors.toSet()));
+        break;
+      case LONG:
+        expressionValues.add(String.valueOf(value.getLong()));
+        break;
+      case LONG_ARRAY:
+        expressionValues.addAll(value.getLongArrayList().stream().map(Object::toString).collect(
+            Collectors.toSet()));
+        break;
+      case DOUBLE:
+        expressionValues.add(String.valueOf(value.getDouble()));
+        break;
+      case DOUBLE_ARRAY:
+        expressionValues.addAll(value.getDoubleArrayList().stream().map(Object::toString).collect(
+            Collectors.toSet()));
+        break;
+      case FLOAT:
+        expressionValues.add(String.valueOf(value.getFloat()));
+        break;
+      case FLOAT_ARRAY:
+        expressionValues.addAll(value.getFloatArrayList().stream().map(Object::toString).collect(
+            Collectors.toSet()));
+        break;
+      case BOOL:
+        expressionValues.add(String.valueOf(value.getBoolean()).toLowerCase());
+        break;
+      case BOOLEAN_ARRAY:
+        expressionValues.addAll(value.getBooleanArrayList().stream()
+            .map(b -> b.toString().toLowerCase()).collect(Collectors.toSet()));
+        break;
+      default:
+        // Ignore the rest of the types for now.
+        throw new IllegalArgumentException("Unsupported value type in subset check.");
+    }
+    return expressionValues;
+  }
+
+  /**
+   * Returns a map from column name to the Filter that's received for that column in the
+   * given query.
+   */
+  @Nonnull
+  private Map<String, Filter> getQueryFilterMap(QueryRequest request) {
```

Comment:
You are not caring if it's an OR or AND filter?

Reply:
I've thought about it. If we handle `OR` and `AND`, the behavior wouldn't change for `AND` but we can actually optimize for `OR` scenario. However, I wanted to avoid over-engineering and keep the logic simple for now but I could be convinced to optimize that though :)

Reply:
Haven't dove into this enough to know if I'm misunderstanding the comment, but OR vs AND seems like it should matter.

A view filtered to `FOO = x` should match a query with `FOO = x AND BAR = y` but can't match `FOO = X OR BAR = y` (since there could be spans with `BAR = Y` not present in this view). Is that distinction accounted for somewhere?

Reply:
@aaron-steinfeld Good point, and that case is not handled with the current change. I was originally debating whether that validation is responsibility of the above layers or the view handler. Handling it here would let the view filter attribute to be treated like any other attribute so I like what you're suggesting.

It sounds like we have to check the entire query tree to make sure there is no `OR` in the hierarchy from the matching view filter node until the root. Is that what you're thinking too?

Reply:
actually, nm about my second sentence. I got the case and I'm working on a fix for that. Thank you!

Reply:
I believe so, although difficult to put into words. More abstractly, any filter tree must be convertible to `<subsetOfviewFilter> && <restOfFilterTree>`.
That could happen in a two different ways, it's a fun tree problem. Recursing from root, a node matches if:
- It's AND and at least one child matches
- It's OR and every child matches

That way:
- Match: `(FOO = x AND BAR = y) OR (FOO = x AND BAR = z)
- No Match: `(FOO = x OR BAR = y) AND (FOO = x OR BAZ = z)

Just for fun... I went down a logic rabbit hole when writing examples, so I guess the above doesn't always hold. Take
`(FOO = x OR BAR = y) AND (FOO = x OR BAR = z)
  ```
(FOO = x OR BAR = y) AND (FOO = x OR BAR = z) -> (un)distribute
(FOO = x ) OR (BAR = y AND BAR = z) -> (mutually exclusive)
(FOO = x ) OR FALSE -> (identity)
FOO = x
  ```
 Not sure how deep we want to go with our boolean logic...

Reply:
Hah missed your second comment because I got carried away, but hopefully my comment is useful, else it was fun to write.

Reply:
Implemented this and handling most (all?) of the cases that we discussed.